libtransmission/rpc-server.cc

```diff
@@ -544,6 +588,108 @@ static auto constexpr ServerStartRetryDelayIncrement = int{ 5 };
 static auto constexpr ServerStartRetryDelayStep = int{ 3 };
 static auto constexpr ServerStartRetryMaxDelay = int{ 60 };
 
+static char const* tr_rpc_address_to_string(tr_rpc_address const* addr, char* buf, size_t buflen)
+{
+    TR_ASSERT(tr_rpc_address_is_valid(addr));
+
+    switch (addr->type)
+    {
+    case TR_RPC_AF_INET:
+        return evutil_inet_ntop(AF_INET, &addr->addr, buf, buflen);
+
+    case TR_RPC_AF_INET6:
+        return evutil_inet_ntop(AF_INET6, &addr->addr, buf, buflen);
+
+    case TR_RPC_AF_UNIX:
+        tr_strlcpy(buf, addr->addr.unixSocketPath, buflen);
+        return buf;
+
+    default:
+        return nullptr;
+    }
+}
+
+static std::string tr_rpc_address_with_port(tr_rpc_server const* server)
+{
+    tr_rpc_address const* addr = server->bindAddress;
+
+    char addr_buf[TrUnixAddrStrLen];
```

Comment:
This seems to suggest that length of the string representation of a UNIX socket "address" is always >= that of IPv4/IPv6 address. While it's seemingly true, I wonder if we still want to use this constant here, given that the buffer is passed into a generic `tr_rpc_address_to_string()` function. Maybe declaring a separate constant would be a better idea.

Reply:
In most platforms I checked, including macOS, this value was above 104 characters, well beyond the size of an ipv6 address. Though, all of this code could be changed to use `std::string` instead, I just copied how it was done in `tr_address_to_string()`

Reply:
So I attempted to implement `tr_address_to_string()` as `std::string` but I ran into some... interesting errors and it made me realize I don't understand the full implications of how `std::string` works when copying local buffers or returning `c_str()`. Therefore, I'd like to leave this for another time.

That being said: https://stackoverflow.com/a/166157 I am still very confident that `TrUnixAddrStrLen` is greater than the max size of an ipv6 address on most platforms.